casepro/profiles/signals.py

```diff
@@ -0,0 +1,20 @@
+from __future__ import unicode_literals
+
+from django.contrib.auth.models import User
+from django.db.models.signals import m2m_changed
+from django.dispatch import receiver
+
+from casepro.msgs.models import Label, Message
+
+from .models import Notification
+
+
+@receiver(m2m_changed, sender=Message.labels.through)
+def notify_label_watchers(sender, instance, action, reverse, model, pk_set, **kwargs):
```

Comment:
Only madness leads down this path! My recommendation would be to make this explicit as opposed to automagic.